### ISSUE TITLE
Outreachy Code Project: Riya Elizabeth John

### ISSUE THREAD

**Issue by Riyabelle25** [ersilia-os/ersilia#159]
Applicant: @Riyabelle25 (she/her)

Welcome to the Ersilia Open Source Initiative. This issue will serve to track all your contributions for the project “Improve the Ersilia Model Hub, a FOSS platform offering pre-trained AI/ML models for research”.

Please tick the tasks as you complete them. To make a final application it is not required to have completed all tasks. This project requires knowledge of the Python programming language. The tasks are not ordered from more to less important, they are simply related to different skills. Start where you feel most comfortable.

---
### Initial steps
- [x] Record your application for the project on the Outreachy website referencing this issue. Please make sure to select the right project on the website.
- [x] Join the Slack channel to follow public communications.
- [x] Comment under this issue explaining why you are interested in this project.
---
### Installation of the Ersilia Model Hub
- [x] Install the `ersilia` library.
- [x] Add a screenshot under this issue showing that you are able to run one model (for example, the chemprop-antibiotic model)
- [x] Fetch at least 3 models from the Ersilia Model Hub. You can find these models with the `ersilia catalog` command. Add a screenshot of the *local* catalog (`ersilia catalog –local`)
---
### CLI
- [x] Check if there are open issues related to the command line interface. Continue with the next tasks if they are open.
- [x] Select one issue related to improving the CLI and request to be assigned to it.
- [x] Link the #PR as a comment under this issue.
- [x] Make any changes required in the PR and tick this box once it has been approved.
- [x] Suggest at least one missing feature in the CLI (one sentence is enough, for example: “Add command to estimate memory usage of a particular model”).
---
### Python library
- [x] Add a screenshot showing that you are able to run predictions using `ersilia` as a Python library (find more information  [here](https://ersilia.gitbook.io/ersilia-book/quick-start/antibiotic-activity-prediction)). Ideally, use a Jupyter notebook.
- [ ] Create a simple [Streamlit](https://streamlit.io/) app using the `ersilia` Python library. The app can have an input and an output box, and perhaps a few models to select. Add a screenshot of the app as seen in your browser.
- [ ] Write a docstring for the ErsiliaModel class. Use the Google Python Style guide. Paste the docstring as a comment below (do not use a PR).
---
### Scientific content
- [x]  Check the models available in the Hub
- [ ] Select one model from the list and write a technical card (what is the model for, what input, which data was used to create it, what kind of ML algorithm uses…) for it
- [ ] Add your card as a comment to this issue
- [ ] Search the scientific literature and suggest 3 new models (comment in this issue) that would be relevant to incorporate in the Hub.
---
### Other
If you have an interest in working on related topics or have new suggestions, please do the following
- [x] Add a comment in this issue with your new idea, tagging the mentor 
- [x] Get feedback from the mentor and act accordingly
- [x] Link in the comments any other PR you have contributed to.
---
### Community
- [x] Look up two other projects and comment on their issues with feedback on one of their tasks
- [x] If you have feedback from your peers, answer it in this issue.
---

### Final application
- [x] I have answered all comments from mentors and contributors
- [x] All PR or issues assigned to me are complete
- [x] I have submitted my final application to the project

**Comment by Riyabelle25** [comment]
Hey there!

I'm Riya Elizabeth John a third-year B.E undergrad at IIT Roorkee, India.
I'm primarily a Python Developer with (both open-source and industrial) experience working in ML Toolboxes/Packages. I love being involved in OpenSource projects and communities- I'd previously contributed to [sktime](https://github.com/alan-turing-institute/sktime) and am currently an MLH (Open-Source) Fellow.
As part of the Vision and Language Group here at my university, I also have a sound knowledge of Deep-learning models and Machine-learning algorithms. 
In short, I really love working on software platforms that intersect with the ML domain, and Ersilia seems to be the perfect fit for me 😊

**Comment by Riyabelle25** [comment]
### Add a screenshot under this issue showing that you are able to run one model (for example, the chemprop-antibiotic model)
<img width="600" alt="Screenshot 2022-03-28 at 1 09 30 PM" src="https://user-images.githubusercontent.com/55790848/161964009-384b93e6-0ab3-4c13-978b-69eb393a8258.png">

**Comment by Riyabelle25** [comment]
### Fetch at least 3 models from the Ersilia Model Hub. You can find these models with the ersilia catalog command. Add a screenshot of the local catalog (ersilia catalog –local
<img width="900" alt="Screenshot 2022-04-06 at 5 30 05 PM" src="https://user-images.githubusercontent.com/55790848/161991016-0361a84f-6f87-467c-970f-dda35615bd1d.png">
)

**Comment by GemmaTuron** [comment]
Hi @Riyabelle25 

Great work thanks! Check the list for some more contribution options. I am also tagging @miquelduranfrigola in case he has suggestions

**Comment by Riyabelle25** [comment]
Thank you @GemmaTuron. 
I had recently built a streamlit app for data visualization (solar flare bursts, from data provided by ISRO), so maybe I can build one for Ersilia as well?

**Comment by Riyabelle25** [comment]
###  Link in the comments any other PR you have contributed to:
1. Currently working on #6 with PR: #188 
2. Also working on #87 with Amna 🙌: Implementing `modelFullDeleter` to delete the unused models

**Comment by Riyabelle25** [comment]
### Add a comment in this issue with your new idea, tagging the mentor:

Seeing contributions #188, #87, #7 all of which result in getting model metadata before/during  `ersilia fetch <model_id>`
My idea is to have an all-purpose `metadata.json` in the root, that stores information such as the `size``, last_used_time` as well as `progress` checkpoints of all of Ersilia's models.
https://github.com/ersilia-os/ersilia/pull/188#issuecomment-109389125

The suggested `METADATA.json`:
![](https://user-images.githubusercontent.com/55790848/162568629-99e4be1c-d000-41db-b4e3-6baf73c07ef1.png)
2

**Comment by Riyabelle25** [comment]
### Add a screenshot showing that you are able to run predictions using ersilia as a Python library

<img width="723" alt="Screenshot 2022-04-10 at 12 44 54 AM" src="https://user-images.githubusercontent.com/55790848/162588524-f5001314-9316-4293-b363-002ba1000b7e.png">

**Comment by Rufaida98** [comment]
@Riyabelle25 your work is very organized 👍

**Comment by Riyabelle25** [comment]
> @Riyabelle25 your work is very organized 👍

@Rufaida98 Thank you! 💛
